src/domains/chart/components/lib-charts/dygraph-chart.tsx

```diff
@@ -515,6 +526,25 @@ export const DygraphChart = ({
         underlayCallback(canvas: CanvasRenderingContext2D, area: DygraphArea, g: Dygraph) {
           updatePrecededPosition(g)
 
+          if (propsRef.current.alarm) {
+            const { alarm: currentAlarm } = propsRef.current
+
+            const alarmPosition = g.toDomXCoord(currentAlarm.when * 1000)
+            const fillColor = getBorderColor(currentAlarm.status)
+
+            if (fillColor) {
```

Comment:
I think that determining the alarm status visibility from the border color is a wrong pattern.
If we do not accept invalid statuses can we make this check when we add the alarm to the state, so we will never have an invalid alarm state?

Reply:
Very good point, i actually already encountered a bug caused by this :) , i'll do exactly what you describe